internal/util/errors.go

```diff
@@ -16,98 +16,46 @@ limitations under the License.
-func (e ErrKeyNotFound) Error() string {
-	return e.err.Error()
-}
-
-// Unwrap returns the encapsulated error.
-func (e ErrKeyNotFound) Unwrap() error {
-	return e.err
-}
-
-// ErrObjectExists is returned when named omap is already present in rados.
-type ErrObjectExists struct {
-	objectName string
-	err        error
-}
-
-// Error returns the error string for ErrObjectExists.
-func (e ErrObjectExists) Error() string {
-	return e.err.Error()
-}
-
-// Unwrap returns the encapsulated error.
-func (e ErrObjectExists) Unwrap() error {
-	return e.err
-}
-
-// ErrObjectNotFound is returned when named omap is not found in rados.
-type ErrObjectNotFound struct {
-	oMapName string
-	err      error
-}
-
-// Error returns the error string for ErrObjectNotFound.
-func (e ErrObjectNotFound) Error() string {
-	return e.err.Error()
-}
-
-// Unwrap returns the encapsulated error.
-func (e ErrObjectNotFound) Unwrap() error {
-	return e.err
-}
-
-// ErrSnapNameConflict is generated when a requested CSI snap name already exists on RBD but with
-// different properties, and hence is in conflict with the passed in CSI volume name.
-type ErrSnapNameConflict struct {
-	requestName string
-	err         error
-}
+import (
+	"errors"
+	"fmt"
+)
 
-// Error returns the error string for ErrSnapNameConflict.
-func (e ErrSnapNameConflict) Error() string {
-	return e.err.Error()
-}
-
-// Unwrap returns the encapsulated error.
-func (e ErrSnapNameConflict) Unwrap() error {
-	return e.err
-}
+var (
+	// ErrKeyNotFound is returned when requested key in omap is not found.
+	ErrKeyNotFound = errors.New("key not found")
+	// ErrObjectExists is returned when named omap is already present in rados.
+	ErrObjectExists = errors.New("object exists")
+	// ErrObjectNotFound is returned when named omap is not found in rados.
+	ErrObjectNotFound = errors.New("object not found")
+	// ErrSnapNameConflict is generated when a requested CSI snap name already exists on RBD but with
+	// different properties, and hence is in conflict with the passed in CSI volume name.
+	ErrSnapNameConflict = errors.New("snapshot name conflict")
+	// ErrPoolNotFound is returned when pool is not found.
+	ErrPoolNotFound = errors.New("pool not found")
+)
 
-// NewErrSnapNameConflict returns a ErrSnapNameConflict error when CSI snap name already exists.
-func NewErrSnapNameConflict(name string, err error) ErrSnapNameConflict {
-	return ErrSnapNameConflict{name, err}
+type errorPair struct {
+	first  error
+	second error
 }
 
-// ErrPoolNotFound is returned when pool is not found.
-type ErrPoolNotFound struct {
-	Pool string
-	Err  error
+func (e errorPair) Error() string {
+	return fmt.Sprintf("%v: %v", e.first, e.second)
 }
 
-// Error returns the error string for ErrPoolNotFound.
-func (e ErrPoolNotFound) Error() string {
-	return e.Err.Error()
+// Is checks if target error is wrapped in the first error.
+func (e errorPair) Is(target error) bool {
+	return errors.Is(e.first, target)
```

Comment:
I think this makes more sense if this is

    return errors.Is(e.first, target) || errors.Is(e.second, target)

Reply:
Then second would be unnecessarily checked twice, because `errors.Is()` would try unwrapping, if `Is()` returns false. We _could_ of course let `Unwrap()` always return `nil`, and let `Is()` check both branches. But we would come to exactly the same behavior with more complicated code. (Because the `Is()` and `Unwrap()` methods are not meant to be called directly, but only from the `errors` package.)